runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -294,11 +294,28 @@ class INTERPRETER_CLASS
 	{
 		U_8 *profilingCursor = startProfilingRecord(REGISTER_ARGS, sizeof(J9Class*) + 2*sizeof(J9Method*));
 		if (NULL != profilingCursor) {
+#ifdef J9ARM
+			// ARMv8 does not support store multiple to unaligned addresses, and unfortunately GCC
+			// generates a store multiple for the normal version of these three assignments.
+			// To avoid the store multiple instructions, we can replace the assignments with
+			// inline assembly code.
+			asm volatile(
```

Comment:
Does this need to be volatile?  Not sure what reordering opportunities would actually occur here.

Reply:
My take is that if I'm putting in inline assembly I want it emitted where I put it, so I use volatile by default. Are there disadvantages to using the keyword?

Reply:
It prevents any reordering across the block, by compiler or processor.

Also here, the cursor appears to not be updated after the asm block.

Reply:
Indeed, the cursor is not used again so does not need updating.

Reply:
I'm not sure volatile would prevent the block itself from being moved, subject to all the usual dependency restrictions.  Certainly there is no compiler reordering within the block, but there could be processor reordering (which would be fine in this case anyway).

Reply:
Right - the cursor has already been updated and stored back before we fill in the data.